newlib/libc/tinystdio/puts.c

```diff
@@ -36,21 +36,28 @@ puts(const char *str)
 {
         int (*put)(char, struct __file *);
```

Comment:
Same question as fputs -- why does this file not follow the same pattern of using a static inline and a wrapper with the locking?

Reply:
Nice catch, thanks!